sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/catalog/SessionCatalog.scala

```diff
@@ -1064,6 +1076,23 @@ class SessionCatalog(
     }
   }
 
+  /**
+   * Make the partition path qualified.
+   * If the partition path is relative, e.g. 'paris', it will be qualified with
+   * parent path using table location, e.g. 'file:/warehouse/table/paris'
+   */
+  private def partitionWithQualifiedPath(
```

Comment:
> Currently only ALTER TABLE t ADD PARTITION(b=1) LOCATION for hive serde table has the expected qualified path. we should make other scenes to be consist with it.

If this is true, can we reuse the related code that is used to qualify paths?

Reply:
https://github.com/apache/spark/blob/c8628354b7d2e6116b2a6eb3bdb2fc957c91fd03/sql/hive/src/main/scala/org/apache/spark/sql/hive/client/HiveClientImpl.scala#L600
Hive client support this.